nexus/src/app/sagas/instance_create.rs

```diff
@@ -60,174 +50,286 @@ pub struct Params {
-            "CreateNetworkInterfaceId",
-            new_action_noop_undo(saga_generate_uuid),
-        );
+// instance create saga: definition
 
-        template_builder.append(
-            format!("network_interface{i}").as_str(),
-            "CreateNetworkInterface",
-            ActionFunc::new_action(
-                async move |sagactx| {
-                    sic_create_network_interface(sagactx, i).await
-                },
-                async move |sagactx| {
-                    sic_create_network_interface_undo(sagactx, i).await
-                },
-            ),
-        );
+#[derive(Debug)]
+pub struct SagaInstanceCreate;
+impl NexusSaga for SagaInstanceCreate {
+    const NAME: &'static str = "instance-create";
+    type Params = Params;
+
+    fn register_actions(registry: &mut super::ActionRegistry) {
+        registry.register(Arc::clone(&*ALLOC_SERVER));
+        registry.register(Arc::clone(&*ALLOC_PROPOLIS_IP));
+        registry.register(Arc::clone(&*CREATE_INSTANCE_RECORD));
+        registry.register(Arc::clone(&*CREATE_NETWORK_INTERFACE));
+        registry.register(Arc::clone(&*CREATE_SNAT_IP));
+        registry.register(Arc::clone(&*CREATE_EXTERNAL_IP));
+        registry.register(Arc::clone(&*CREATE_DISKS_FOR_INSTANCE));
+        registry.register(Arc::clone(&*ATTACH_DISKS_TO_INSTANCE));
+        registry.register(Arc::clone(&*INSTANCE_ENSURE));
     }
 
-    // Allocate an external IP address for the default outbound connectivity
-    template_builder.append(
-        "snat_ip_id",
-        "CreateSnatIpId",
-        new_action_noop_undo(saga_generate_uuid),
-    );
-    template_builder.append(
-        "snat_ip",
-        "CreateSnatIp",
-        ActionFunc::new_action(
-            sic_allocate_instance_snat_ip,
-            sic_allocate_instance_snat_ip_undo,
-        ),
-    );
+    fn make_saga_dag(
+        params: &Self::Params,
+        mut builder: steno::DagBuilder,
+    ) -> Result<steno::Dag, SagaInitError> {
+        let instance_id = Uuid::new_v4();
+
+        builder.append(Node::constant(
+            "instance_id",
+            serde_json::to_value(&instance_id).map_err(|e| {
+                SagaInitError::SerializeError(String::from("instance_id"), e)
+            })?,
+        ));
+
+        builder.append(Node::action(
+            "propolis_id",
+            "GeneratePropolisId",
+            ACTION_GENERATE_ID.as_ref(),
+        ));
+
+        builder.append(Node::action(
+            "server_id",
+            "AllocServer",
+            ALLOC_SERVER.as_ref(),
+        ));
+
+        builder.append(Node::action(
+            "propolis_ip",
+            "AllocatePropolisIp",
+            ALLOC_PROPOLIS_IP.as_ref(),
+        ));
+
+        builder.append(Node::action(
+            "instance_name",
+            "CreateInstanceRecord",
+            CREATE_INSTANCE_RECORD.as_ref(),
+        ));
+
+        // Helper function for append subsagas to our parent saga.
```

Comment:
s/for append/to append/ ?

Reply:
Fixed in 33951e28e39cfd0dfeacf5066700dff6f206b045.